integration/query_array_compat_test.go

```diff
@@ -84,12 +84,14 @@ func testQueryArrayCompatDotNotation() map[string]queryCompatTestCase {
 			resultType: emptyResult,
 		},
 		"Field": {
-			filter:        bson.D{{"v.array", int32(42)}},
-			skipForTigris: "Tigris does not support language keyword 'array' as field name",
+			filter:         bson.D{{"v.array", int32(42)}},
+			skipForTigris:  "Tigris does not support language keyword 'array' as field name",
+			resultPushdown: true,
```

Comment:
We have `"array-documents": bson.A{bson.D{{"field", int32(42)}}, bson.D{{"field", int32(44)}}},` in `Composites`. This might be interesting because we might find out if array -> object -> array doesn't work https://github.com/FerretDB/FerretDB/issues/1828 or array -> object doesn't work.

 
Could you try to push down to match one of the field like ` int32(42)`? I think query would be like this, but please double check. :hugs: 
```
			filter:         bson.D{{"v.array.0.field", int32(42)}},
```

If it doesn't work we can update the issue 1828 :hugs:

Reply:
Well, `v.array.0.field` will work in the current PR, but `v.array.field` is for #1828, yeah. 

I see there is `v.array.0.foo` case listed in this file, it covers this logic. I don't think we need any extra cases.